pkg/select/blockdevice/filters.go

```diff
@@ -242,3 +250,66 @@ func filterNodeName(originalBD *apis.BlockDeviceList, spec *apis.DeviceClaimSpec
 	}
 	return filteredBDList
 }
+
+// filterBlockDeviceTag is used to filter out BlockDevices which do not have the
+// block-device-tag label. This filter works on a block device list which has
+// already been filtered by the given selector.
+func filterBlockDeviceTag(originalBD *apis.BlockDeviceList, spec *apis.DeviceClaimSpec) *apis.BlockDeviceList {
+
+	// if the block-device-tag label was already included in the selector
+	// given in the BDC by the user, then this filter is not required. This
+	// is because it would have already performed the filter operation with the
+	// label. If the label is not present, a new selector is made to remove
+	// devices which have that label.
+	if !isBlockDeviceTagSelectorRequired(spec.Selector) {
+		return originalBD
+	}
+
+	// a DoesNotExist requirement is created to filter out devices which have
+	// the block-device-tag label
+	blockDeviceTagRequirement, err := labels.NewRequirement(kubernetes.BlockDeviceTagLabel, selection.DoesNotExist, []string{})
+
+	// this error should never occur, because error for DoesNotExist happen
+	// only when non zero length of values are passed
+	if err != nil {
+		klog.Info("could not create requirement for label ", kubernetes.BlockDeviceTagLabel)
+		return originalBD
+	}
+
+	blockDeviceTagDoesNotExistSelector := labels.NewSelector()
+	blockDeviceTagDoesNotExistSelector.Add(*blockDeviceTagRequirement)
+
+	filteredBDList := &apis.BlockDeviceList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "BlockDevice",
+			APIVersion: "openebs.io/v1alpha1",
+		},
+	}
+
+	for _, bd := range originalBD.Items {
+		// if the tag label is not present, the BD will be included in the list
+		if blockDeviceTagDoesNotExistSelector.Matches(labels.Set(bd.Labels)) {
+			filteredBDList.Items = append(filteredBDList.Items, bd)
+		}
+	}
+	return filteredBDList
+}
+
+// isBlockDeviceTagSelectorRequired is used to check whether a selector
+// was present on the BDC. It is used to decide whether a does not exist selector
+// for the block-device-tag label should be applied or not.
+func isBlockDeviceTagSelectorRequired(options *metav1.LabelSelector) bool {
```

Comment:
function name seems to be reverse of what it is supposed to do

Reply:
Done.